tests/cupy_tests/sorting_tests/test_sort.py

```diff
@@ -437,3 +437,129 @@ def test_partition_invalid_negative_axis(self, xp):
         kth = 2
         axis = -4
         return self.partition(a, kth, axis=axis)
+
+
+@testing.parameterize(*testing.product({
+    'external': [False, True],
+}))
+@testing.gpu
+class TestArgpartition(unittest.TestCase):
+
+    _multiprocess_can_split_ = True
+
+    def argpartition(self, a, kth, axis=-1):
+        if self.external:
+            xp = cupy.get_array_module(a)
+            return xp.argpartition(a, kth, axis=axis)
+        else:
+            return a.argpartition(kth, axis=axis)
+
+    # Test base cases
+
+    @testing.numpy_cupy_raises()
+    def test_argpartition_zero_dim(self, xp):
+        a = testing.shaped_random((), xp)
+        kth = 2
+        return self.argpartition(a, kth)
+
+    @testing.for_all_dtypes(no_float16=True, no_bool=True)
+    @testing.numpy_cupy_equal()
+    def test_argpartition_one_dim(self, xp, dtype):
+        a = testing.shaped_random((10,), xp, dtype, 100)
+        kth = 2
+        idx = self.argpartition(a, kth)
+        return idx[kth]
```

Comment:
Can you check its values? I mean `(a[idx[:kth]] < a[idx[kth]]).all()` and `(a[idx[kth]] < a[idx[kth:]]).all()`. Please check it for all test cases.